### PR TITLE
Task manager int tests

### PR DESCRIPTION
**Pull request by JoeBatt1989**

### CODE REVIEW

File: tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                file.RelativeRootPath.Should().Be(taskDispatchFile?.RelativeRootPath);
+            }
+            Output.WriteLine("Details of ClinicalReviewRequestEvent matches TaskDispatchEvent");
+        }
+
+        public void AssertTaskUpdateEventFromTaskCallback(TaskUpdateEvent taskUpdateEvent, TaskCallbackEvent taskCallbackEvent, TaskExecutionStatus status)
+        {
+            Output.WriteLine("Asserting details of TaskUpdateEvent with TaskCallbackEvent");
+            taskUpdateEvent.ExecutionId.Should().Be(taskCallbackEvent.ExecutionId);
+            // BUG taskUpdateEvent.CorrelationId.Should().Be(taskDispatchEvent.CorrelationId);
```
Comment (RemakingEden) [inline tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs:48]: There are a few commented out lines in this document. Just checking these are expected?

File: tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            }
+
+            Output.WriteLine("Details of TaskUpdateEvent matches TaskCallbackEvent");
+        }
+
+        public void AssertTaskUpdateEventFromTaskDispatch(TaskUpdateEvent taskUpdateEvent, TaskDispatchEvent taskDispatchEvent, TaskExecutionStatus status)
+        {
+            Output.WriteLine("Asserting details of TaskUpdateEvent with TaskDispatchEvent");
+            taskUpdateEvent.ExecutionId.Should().Be(taskDispatchEvent.ExecutionId);
+            // BUG taskUpdateEvent.CorrelationId.Should().Be(taskDispatchEvent.CorrelationId);
```
Comment (RemakingEden) [inline tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs:67]: here

File: tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            clinicalReviewRequestEvent.PatientMetadata.PatientDob.Should().Be(GetTaskPluginArguments(taskDispatchEvent, "patient_dob"));
+            clinicalReviewRequestEvent.WorkflowName.Should().Be(GetTaskPluginArguments(taskDispatchEvent, "workflow_name"));
+
+            foreach (var file in clinicalReviewRequestEvent.Files)
+            {
+                var taskDispatchFile = taskDispatchEvent.Inputs.FirstOrDefault(x => x.Name.Equals(file.Name));
+
+                file.Name.Should().Be(taskDispatchFile?.Name);
+                file.Endpoint.Should().Be(taskDispatchFile?.Endpoint);
+                // file.Credentials.Should().BeEquivalentTo(taskDispatchFile?.Credentials); - WorkflowManager to pass the credentials
```
Comment (RemakingEden) [inline tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs:37]: here

**Review by RemakingEden — CHANGES_REQUESTED** [review]

File: tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs (hunk 1 above)
Comment (JoeBatt1989) [reply tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs:48]: Yeah theres a bug open for this.

File: tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs (hunk 2 above)
Comment (JoeBatt1989) [reply tests/IntegrationTests/TaskManager.IntegrationTests/Support/Assertions.cs]: Bug 227 open for this issue.

File: src/Common/TypeExtensions.cs [hunk 4]
```diff
@@ -38,7 +39,14 @@ public static Type GetType(this Type interfaceType, string typeString)
                       typeString,
                       (name) =>
                       {
-                          return AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(z => !string.IsNullOrWhiteSpace(z.FullName) && z.FullName.StartsWith(name.FullName));
+                          var assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(z => !string.IsNullOrWhiteSpace(z.FullName) && z.FullName.StartsWith(name.FullName));
+
+                          if (assembly is null)
+                          {
+                              assembly = Assembly.LoadFile($"{AppDomain.CurrentDomain.BaseDirectory}{name.FullName}.dll");
```
Comment (jackschofield23) [inline src/Common/TypeExtensions.cs:46]: src/Common/TypeExtensions.cs#L46

Replace this call to 'Assembly.LoadFile' with 'Assembly.Load'.

Comment (jackschofield23) [reply src/Common/TypeExtensions.cs:46]: Might be worth doing this change

File: tests/IntegrationTests/TaskManager.IntegrationTests/Hooks.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            ObjectContainer.RegisterInstanceAs(TaskUpdateConsumer, "TaskUpdateConsumer");
+            ObjectContainer.RegisterInstanceAs(ClinicalReviewConsumer, "ClinicalReviewConsumer");
+            ObjectContainer.RegisterInstanceAs(MinioClient);
+            var dataHelper = new DataHelper(ObjectContainer);
+            ObjectContainer.RegisterInstanceAs(dataHelper);
+        }
+
+        //[BeforeTestRun(Order = 1)]
+        //[AfterTestRun(Order = 0)]
+        //public async static Task ClearTestData()
```
Comment (jackschofield23) [inline tests/IntegrationTests/TaskManager.IntegrationTests/Hooks.cs:126]: this meant to be left in/used?

**Review by jackschofield23 — APPROVED** [review]

File: tests/IntegrationTests/TaskManager.IntegrationTests/Hooks.cs (hunk 5 above)
Comment (JoeBatt1989) [reply tests/IntegrationTests/TaskManager.IntegrationTests/Hooks.cs]: Removed

**Review by RemakingEden — APPROVED** [review]

**Pull request closed without merging**